lib/working_hours/config.rb

```diff
@@ -115,7 +116,7 @@ def compile_time time
         sec = time[TIME_FORMAT,3].to_i
         time = hour * 3600 + min * 60 + sec
         # Converts 24:00 to 23:59:59.999999
-        return 86399.999999 if time == 86400
+        return MIDNIGHT if time == 86400
```

Comment:
Using a Rational instead of a floating number (`Time` can actually be stored as Rational internally for infinite precision)
Before that there was a lot of random digits at the end of the floating part.